src/key.cpp

```diff
@@ -275,6 +275,13 @@ bool CKey::SignSchnorr(const uint256& hash, Span<unsigned char> sig, const uint2
         if (!secp256k1_keypair_xonly_tweak_add(GetVerifyContext(), &keypair, tweak.data())) return false;
     }
     bool ret = secp256k1_schnorrsig_sign(secp256k1_context_sign, sig.data(), hash.data(), &keypair, aux ? (unsigned char*)aux->data() : nullptr);
+#ifdef ENABLE_SCHNORR_VERIFY
+    if(ret){
+        secp256k1_xonly_pubkey pubkey_verify;
+        ret = secp256k1_keypair_xonly_pub(GetVerifyContext(), &pubkey_verify, nullptr, &keypair);
+        ret &= secp256k1_schnorrsig_verify(GetVerifyContext(), sig.data(), hash.begin(), 32, &pubkey_verify);
+    }
+#endif
```

Comment:
For API safeness it's probably better to clear the output `sig` if ret is `false`.

Reply:
Like [this](https://github.com/bitcoin/bitcoin/pull/22934/commits/c660374a37f4a83a774db55e15e4e67d1d33ef3c)?

Reply:
How about using `memory_cleanse()`.

Reply:
thanks